specifications/api/swagger.yaml

```
@@ -1088,6 +1116,35 @@ x-transactionWithdrawals: &transactionWithdrawals
       stake_address: *stakeAddress
       amount: *amount
 
+x-softIndex: &softIndex
+  type: integer
+  minimum: 0
+  maximum: 2147483647
+  description: A soft derivation index.
+
+x-multiDelegationAction: &multiDelegationAction
+  description: |
+    A delegation action for a given stake key denoted by its soft index.
+    Pool id is only required for "join".
+    Stake key index are required for both actions.
+  type: object
+  required:
+    - action
+    - stake_key_index
+  properties:
+    action:
+      type: string
+      enum: ["quit", "join"]
+    pool: *stakePoolId
+    stake_key_index: *derivationSegment
```

What do you think of this way? I like it more.

```suggestion
  oneOf:
    - type: object
      required:
        - join
      properties:
        join:
          type: object
          required:
            - pool
            - stake_key_index
          properties:
            pool: *stakePoolId
            stake_key_index: *derivationSegment
    - type: object
      required:
        - quit
      properties:
        quit:
          type: object
          required:
            - stake_key_index
          properties:
            stake_key_index: *derivationSegment
```

---

I agree it is better. Adopted